iocore/net/quic/QUICStreamState.h

```diff
@@ -65,94 +70,93 @@ class QUICStreamState
   virtual void update_with_sending_frame(const QUICFrame &frame)   = 0;
   virtual void update_with_receiving_frame(const QUICFrame &frame) = 0;
 
+  virtual void
+  update_on_transport_send_event()
```

Comment:
These three sound ATS specific. As you can see on the charts, QUIC stream state changes when it received frames, when an application read data, or when an application read RST.

I know `QUICStream` class currently doesn't call update_x at right places, but I don't think we need another update_x.